File: javascript/217.Contains Duplicate.js
let nums = [1, 2, 3, 1]
let nums2 = [1, 1, 1, 3, 3, 4, 3, 2, 4, 2]
let nums3 = [1, 2, 3, 4]


// function solution(array){

//     let result = false
//     //console.log(array)

//     let sorted = array.sort()
//     //console.log(sorted)

//     for(let i =0; i < array.length ; i++){
//         for(let j= i+1; j< array.length ; j++){

//             if(array[i]=== array[j]){
//                 result = true
//             }

//         }
//     }

//     return result
// }

function containsDuplicate(array) {
    let checked = new Set(array)

    let result = false
    if (checked.size === array.length) {
        result = false
    } else {
        result = true
    }
    return result

}

console.log(containsDuplicate(nums))
console.log(containsDuplicate(nums2))
console.log(containsDuplicate(nums3))

File: neetcode/217. Contains Duplicate.py
from collections import Counter
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        checker = True
        counts = Counter(nums)
        result = counts == len(nums)
        
        for x in counts.values(): 
            if x > 1:
                return True``
        return False

        # for i in range(length - 1):
        #     if i == length:
        #         if nums[i] == nums[-1]:
        #             chekcer = False
        #             return checker
        #     else:
        #         if nums[i]==nums[i+1]:
        #                 chekcer = False
        #                 return checker

        # There should be way to loop and check each elements
        # counts function in python will give you key : value    
        # if  [1 1 2 2 2 ] =>  [1:2 , 2:3]
        #  counts.keys : 1 ,2    counts.values: 2,3 
            
        
        